### DISCUSSION TITLE
Having trouble understanding how I can do multiple expressions to search in the gui

### DISCUSSION THREAD

**Discussion by sheldonhull** [junegunn/fzf#3293] [Q&A]
```shell
echo '["foo", "bar", "foobar", "taco", "burrito"]' | jq -r '.[]' | fzf -m | jq -nR '[inputs | select(length>0)]'
```

I've tried various examples from other discussions and help, but can't figure out why the filtering won't let me do a combination of search terms. 

For example, I'd want to do `foo|burrito`, `foo OR burrito`, `(foo)|(burrito)`, `(.*foo$)|(^burrito$)` or something like that. I can't get anything to work other than a simple single search term.

The goal I had was to pipe input in from json, use fzf for a multiselection with tab, and then pipe to the next step. Not sure what I'm doing wrong, or if fzf is the wrong tool for this type of usage?

**Comment by tg0h** [comment]
try `foo | burrito` not `foo|burrito`

**Comment by LangLangBart** [comment]
- see [ADVANCED.md](https://github.com/junegunn/fzf/blob/master/ADVANCED.md#switching-between-ripgrep-mode-and-fzf-mode) for more details and the `man` page for `fzf` for the used elements


```sh
hull() {
	INITIAL_QUERY="${*:-}"
	INPUT_LIST=$'echo \'["foo", "bar", "foobar", "taco", "burrito"]\' | jq -r \'.[]\' | grep -E '
	FZF_DEFAULT_COMMAND="$INPUT_LIST $(printf %q "$INITIAL_QUERY")" \
		fzf --multi --query "$INITIAL_QUERY" --disabled \
		--bind "change:first+reload-sync:$INPUT_LIST {q} || true" | jq -nR '[inputs | select(length>0)]'
}
```

![them](https://github.com/junegunn/fzf/assets/92653266/f438efc4-fa2a-4fe5-8c14-719ca5c852f0)
